packages/mask/src/plugins/Tips/components/Guide.tsx

```diff
@@ -0,0 +1,194 @@
+import { useI18N } from '../locales/index.js'
+import { finishUserGuide, TIPS_GUIDE_TOTAL, useTipsUserGuide } from '../storage/index.js'
+import type { EnhanceableSite } from '@masknet/shared-base'
+
+const useStyles = makeStyles()((theme) => ({
+    container: {
+        position: 'absolute',
+        borderRadius: '50%',
+    },
+    target: {
+        background: 'transparent',
+    },
+    mask: {
+        position: 'fixed',
+        top: 0,
+        width: '100vw',
+        height: '100vh',
+        background: 'transparent',
+        zIndex: 1000,
+    },
+    card: {
+        position: 'absolute',
+        left: 0,
+        width: 256,
+        padding: '16px',
+        borderRadius: '16px',
+        background: theme.palette.maskColor.tips,
+        boxShadow: '0 4px 8px rgba(0,0,0,.1)',
+        boxSizing: 'border-box',
+        color: theme.palette.maskColor.bottom,
+        '&.arrow-top:after': {
+            content: '""',
+            display: 'inline-block',
+            width: 0,
+            height: 0,
+            border: 'solid 8px transparent',
+            borderBottomColor: theme.palette.maskColor.tips,
+            borderBottomWidth: '13px',
+            borderTopWidth: 0,
+            position: 'absolute',
+            top: '-13px',
+            left: '24px',
+        },
+        '&.arrow-bottom:after': {
+            content: '""',
+            display: 'inline-block',
+            width: 0,
+            height: 0,
+            border: 'solid 8px transparent',
+            borderTopColor: theme.palette.maskColor.tips,
+            borderTopWidth: '13px',
+            borderBottomWidth: 0,
+            position: 'absolute',
+            bottom: '-13px',
+            left: '24px',
+        },
+    },
+    buttonContainer: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        paddingTop: '16px',
+    },
+    button: {
+        width: '100%',
+        borderRadius: '20px',
+    },
+}))
+
+export interface GuideStepProps extends PropsWithChildren<{}> {
+    arrow?: boolean
+    disabled?: boolean
+    onComplete?: () => void
+}
+
+export default function Guide({ children, arrow = true, disabled = false, onComplete }: GuideStepProps) {
+    const t = useI18N()
+    const { classes, cx } = useStyles()
+    const childrenRef = useRef<HTMLElement>()
+    const [clientRect, setClientRect] = useState<any>({})
+    const [open, setOpen] = useState(false)
+    const [bottomAvailable, setBottomAvailable] = useState(true)
+    const ui = activatedSocialNetworkUI
+    const enableUserGuide = ui.configuration.tipsConfig?.enableUserGuide
+    const lastStep = useTipsUserGuide(ui.networkIdentifier as EnhanceableSite)
+
+    useEffect(() => {
+        if (disabled || !enableUserGuide || lastStep.finished) return
+        setOpen(true)
+    }, [])
+
+    useEffect(() => {
+        if (disabled || !enableUserGuide || lastStep.finished) return
+        document.body.style.overflow = open ? 'hidden' : ''
```

Comment:
We can add a shared component for resuing the Guide logic.

Reply:
The  mirror tips guide logic have difference with extension setup guide.

Reply:
No, they should share the same logic. CC @nuanyang233.

Reply:
Maybe we can add `Tip Guide` to plugin definition.

Reply:
https://mask.atlassian.net/browse/MF-2215